### PR TITLE
Add deposit workflow thread components

### PR DESCRIPTION
**Pull request by burieberry**
To Do:
- [x] Add tests for `compare-timestamps` helper and revise the helper accordingly
- [x] Update workflow messages with text from updated scenario

header
deposit-wf

### CODE REVIEW

File: packages/web-client/app/components/card-pay/deposit-workflow/index.hbs [hunk 1]
```diff
@@ -1,26 +1,92 @@
...
             {{#if postable.message}}
-              {{format-workflow-message postable.message}}
+              <Boxel::ThreadMessage
+                class="deposit-workflow__thread-message"
+                @datetime={{postable.timestamp}}
+                @imgURL={{postable.author.imgURL}}
+                @hideMeta={{postable-meta-identical postable (object-at milestone.visiblePostables (dec j))}}
+                data-test-postable={{j}} data-test-milestone={{i}}
+              >
+                {{format-workflow-message postable.message}}
```
Comment (Aierie) [inline packages/web-client/app/components/card-pay/deposit-workflow/index.hbs:47]: image

image

image

image

Some browser inconsistencies with list styling - top is Safari, bottom is Chrome. I think using user-agent defaults is ok for now, but ideally we should keep the reset we had before and scope the styles for the `format-workflow-message` styles to a "wrapper" selector, eg. `.wrapper ul`.

File: packages/web-client/app/components/card-pay/deposit-workflow/index.hbs [hunk 2]
```diff
@@ -1,26 +1,92 @@
...
-          <div class="thread-message" data-test-thread-message={{j}} data-test-milestone={{i}}>
-            <div data-test-author-name>
-              {{postable.author.name}}
-            </div>
+<Boxel::Thread class="deposit-workflow">
+  <:header>
+    <Boxel::ThreadHeader @title={{this.workflow.name}} />
+  </:header>
+  <:content>
+    <Boxel::ExpandableBanner
```
[the remark this answers is not in the document]
Comment (Aierie) [reply packages/web-client/app/components/card-pay/deposit-workflow/index.hbs:6]: https://github.com/cardstack/boxel/pull/342

File: packages/web-client/app/components/card-pay/deposit-workflow/index.hbs (hunk 1 above)
Comment (lukemelia) [reply packages/web-client/app/components/card-pay/deposit-workflow/index.hbs:47]: What do you think about a "prose" or "standard-formatting" utility class that we apply to elements that contain markdown-rendered content.

Comment (Aierie) [reply packages/web-client/app/components/card-pay/deposit-workflow/index.hbs:47]: I think that fits nicely!

File: packages/web-client/app/components/card-pay/header-nav/index.hbs [hunk 3]
```diff
@@ -1,9 +1,9 @@
 <nav class="card-pay-header-nav">
-  <ul class="card-pay-header-nav__list">
+  <ul class="card-pay-header-nav__list" role="tablist">
     {{#each @tabs as |tab i|}}
       <li>
-        <LinkTo @route={{tab.route}} class="card-pay-header-nav__tab" data-test-card-pay-header-tab={{i}}>
-          <span class="card-pay-header-nav__tab-icon" />
+        <LinkTo role="tab" id={{tab.route}} @route={{tab.route}} class="card-pay-header-nav__tab" data-test-card-pay-header-tab={{i}}>
+          {{svg-jar tab.icon width="1rem" height="1rem" aria-hidden="true"}}
```
[the remark this answers is not in the document]
Comment (lukemelia) [reply packages/web-client/app/components/card-pay/header-nav/index.hbs]: ✅

File: packages/web-client/app/components/card-pay/deposit-workflow/index.hbs (hunk 1 above)
Comment (lukemelia) [reply packages/web-client/app/components/card-pay/deposit-workflow/index.hbs]: ✅

File: packages/web-client/app/components/card-pay/deposit-workflow/index.hbs (hunk 2 above)
[the remark this answers is not in the document]
Comment (lukemelia) [reply packages/web-client/app/components/card-pay/deposit-workflow/index.hbs:6]: Will pick this up with next boxel update

**Pull request merged**